korge/src/commonMain/kotlin/com/soywiz/korge/view/SpriteAnimation.kt

```diff
@@ -44,5 +39,11 @@ class SpriteAnimation(
         }
     }
 
-    fun getSprite(index : Int) = spriteStack[(index % spriteStack.size)]
+    fun getSprite(index : Int) : BmpSlice = spriteStack[index.realMod(spriteStack.size)]
+}
+
+// Kotlin % does produce mathematically wrong values vor % of negative Numbers
+fun Int.realMod(y: Int): Int {
```

Comment:
Usually operator-like functions include the `infix` operator to not require dot and parenthesis when calling.
Not sure if exposed by any of korlibs. Do you have access to `umod` or was it internal? That function supports mod with negative values like this one